### PR TITLE
upgrade to polkadot 1.0

### LINKED ISSUE #275: Upgrade to Polkadot 1.0
upgrades subtensor to polkadot 1.0.0

AC:

- [x] breaking changes for 0.9.39 => 1.0 (tons of stuff, but one bullet point now lol)
- [X] revamp CI, now broken into 6 separate jobs, takes no more than ~6-7 mins to complete, cargo fmt feedback in ~22 seconds, `cargo check --workspace` in about 3 mins 😎, also added CI checks for clippy and one that asserts that `cargo fix` has no trivial fixes available
- [x] `cargo check` passing
- [x] `cargo check --workspace` passing
- [x] `cargo test` _compiling_
- [x] `cargo test --workspace` _compiling_
- [x] `cargo test --workspace` passing
- [x] `cargo test --workspace --features=runtime-benchmarks` _compiling_
- [x] `cargo test --workspace --features=runtime-benchmarks` passing

### PR DESCRIPTION
**Pull request by sam0x17**
- [x] breaking changes for 0.9.39 => 1.0 (tons of stuff, but one bullet point now lol)
- [X] revamp CI, now broken into 6 separate jobs, takes no more than ~6-7 mins to complete, cargo fmt feedback in ~22 seconds, `cargo check --workspace` in about 3 mins 😎, also added CI checks for clippy and one that asserts that `cargo fix` has no trivial fixes available
- [x] `cargo check` passing
- [x] `cargo check --workspace` passing
- [x] `cargo test` _compiling_
- [x] `cargo test --workspace` _compiling_
- [x] `cargo test --workspace` passing
- [x] `cargo test --workspace --features=runtime-benchmarks` _compiling_
- [x] `cargo test --workspace --features=runtime-benchmarks` passing

fixes #275

### CODE REVIEW

File: node/src/service.rs [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+    }
+
+    fn register_static<T>(_registry: &mut T) -> core::result::Result<(), T::Error>
+    where
+        T: HostFunctionRegistry,
+    {
+        Ok(())
+    }
+}
+
```
Comment (sam0x17) [inline node/src/service.rs:32]: I wonder if this one might be causing any issues? probably not but leaving this flagged _just in case_

File: node/src/service.rs [hunk 2]
```diff
@@ -46,24 +62,18 @@ pub fn new_partial(
...
         ),
     >,
     ServiceError,
 > {
-    if config.keystore_remote.is_some() {
-        return Err(ServiceError::Other(
-            "Remote Keystores are not supported.".into(),
-        ));
-    }
-
```
Comment (sam0x17) [inline node/src/service.rs:66]: `config.keystore_remote` was removed, so we don't need this block anymore AFAIK

File: node/src/service.rs [hunk 3]
```diff
@@ -165,24 +175,15 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
...
-        match remote_keystore(url) {
-            Ok(k) => keystore_container.set_remote_keystore(k),
-            Err(e) => {
-                return Err(ServiceError::Other(format!(
-                    "Error hooking up remote keystore for {}: {}",
-                    url, e
-                )))
-            }
-        };
-    }
```
Comment (sam0x17) [inline node/src/service.rs:184]: `config.keystore_remote` was completely removed, so no need for this now

File: pallets/subtensor/tests/staking.rs [hunk 4]
```diff
@@ -847,7 +847,7 @@ fn test_remove_balance_from_coldkey_account_ok() {
         // Should be able to withdraw without hassle
         let result =
             SubtensorModule::remove_balance_from_coldkey_account(&coldkey_account_id, ammount);
-        assert_eq!(result, true);
+        assert!(result.is_ok());
```
Comment (sam0x17) [inline pallets/subtensor/tests/staking.rs:1068]: why shake your gift when you can open it and see if it's broken directly ;)

```suggestion
        result.unwrap();
```

remember this is all about "what will the dev ux be for the programmer who encounters the failing case when some change makes this start failing"

**Review by keithtensor — APPROVED** [review]

**Comment by shibshib** [comment]
Don't forget to update spec version to 145

File: pallets/subtensor/tests/root.rs [hunk 5]
```diff
@@ -649,7 +653,7 @@ fn test_weights_after_network_pruning() {
         );
         log::info!("Max subnets: {:?}", SubtensorModule::get_max_subnets());
         let i = (n as u16) + 1;
-        let hot: U256 = U256::from(i);
+        let _hot: U256 = U256::from(i);
```
Comment (sam0x17) [inline pallets/subtensor/tests/root.rs]:
```suggestion
        // let _hot: U256 = U256::from(i);
```

File: rust-toolchain.toml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2023-03-01"
+channel = "nightly-2024-03-05"
 components = [ "rustfmt" ]
 targets = [ "wasm32-unknown-unknown" ]
 profile = "minimal"
```
Comment (sam0x17) [inline rust-toolchain.toml]:
```suggestion
profile = "minimal"

```

**Review by sam0x17 — COMMENTED** [review]
giving this my blessing as I can't actually approve

**Review by open-junius — APPROVED** [review]
LGTM!

**Review by distributedstatemachine — APPROVED** [review]
LGTM!

**Pull request merged**